README.md

```diff
@@ -1,108 +1,108 @@
+decisions, ideas, algorithms, APIs, practices and so on, distill them to bare
+minimum and reimplement them or adapt to Mimiker code base. We hope not to
+repeat their mistakes and move away from legacy and non-perfect solutions.
+
+Mimiker project gathers like minded people who value minimalism, simplicity and
+readability of code. We strive for the lowest possible complexity of solutions.
+We love to throw away code that isn't terribly useful or handles rare edge
+cases. We know value of debuggability and we don't hesitate to spend time
+writing tools that help to improve it.
+
+Though userspace programs are part of Mimiker project, they've got simply ported
+from NetBSD or [suckless][1] project. We focus on kernel development, since we
+find it more interesting. We don't want to invest too much time into the device
+drivers, so we keep a list of target platforms small.
+
+If you'd like to get involved in the project please read our [Wiki][2] to find
+out more!
+
+## Where we are?
+
+Mimiker is a real-time operating system. The kernel is preemptible and our
+mutexes support priority inheritance. We minimize work done in interupt context
+by delegating it to interrupt threads instead of running it using soft
+interrupts.
+
+Mimiker runs on [MIPS][15] (32-bit), [AArch64][9] and [RISC-V][10] (both 32-bit
+and 64-bit) architectures under [QEmu][11] and [Renode][12] control.
+
+Mimiker has nice set of debugging tools: `gdb` scripts written in Python, Kernel
+Address Sanitizer, Lock dependency validator, Kernel Concurrency Sanitizer. We
+even have support for profiling the kernel using `gprof`! We use [Clang][19] to
+compile our code base, hence we can employ sophisticated dynamic and static
+analysis algorithms to aid code reliablity.
+
+A common set of synchronization primitives is provided, i.e. spin-locks, mutexes
+and conditional variables - all with simple semantics. We don't have multiple
+primitives that do similar things, but a little bit differently, which is common
+for FreeBSD or Linux kernels.
+
+Mimiker's kernel memory is wired (i.e. non-swappable), so you don't have to
+worry about choosing right locks when accessing kernel memory, unlike in
+FreeBSD.  We have buddy memory allocator for physical memory, virtual address
+space allocator and slab allocator based on [Magazines and Vmem][3] paper. Our
+memory allocators are simple yet efficient.
+
+Mimiker's driver infrastructure abstracts away concept of hardware register
+and interrupts in similar manner to FreeBSD's [NewBus][14]. Special care is
+taken to make drivers portable. We have enumerator routines that autodetect
+devices attached to PCI and USB buses. We use [flat device tree][13] to drive
+kernel configuration during startup phase.
+
+Virtual file system and user virtual address space management are loosely based
+on FreeBSD ideas. They need substatial amount of work to become as mature as in
+FreeBSD or Linux kernels.
+
+## What we're proud of ?
```

Comment:
What we're proud of ? -> What we're proud of?
I'm not sure which looks better

Reply:
IMHO space before '?' looks bad. Again, shouldn't it be "What are we proud of?", or simply "What we are proud of" (without question mark).